### PR TITLE
Rotatable Tip

### LINKED ISSUE #1726: Rotatable Tip
- one instance
- dictionary of tip
- dummy rotating

### PR DESCRIPTION
**Pull request by DenisVorop**
## PR includes
- [x] Feature

## Related issues
Resolve #1726 

## QA Instructions, Screenshots, Recordings
  

https://github.com/taskany-inc/issues/assets/95316053/b7521ff3-3d73-48cd-9ce2-0ac5a3099363

### CODE REVIEW

File: src/components/RotatableTip/RotatableTip.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    };
+
+    const tips: ((ctx?: TipContext) => ReactNode)[] = [
+        () =>
+            generateTip('Hold {key} to see available hotkeys', {
+                key: <Keyboard key="h">h</Keyboard>,
+            }),
+    ];
+
+    const contextTips: Record<TipContext, ((ctx?: TipContext) => ReactNode)[]> = {
```
Comment (awinogradov) [inline src/components/RotatableTip/RotatableTip.tsx:29]: Have u ever heard about memoization?:)

File: src/components/RotatableTip/RotatableTip.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        goal: [
+            (ctx) => generateTip('Press {key} to create {entity}', { ctx }),
```
Comment (awinogradov) [inline src/components/RotatableTip/RotatableTip.tsx:31]: I think we can use `Press {key} to save {entity}` for both cases.

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Review by awinogradov — APPROVED** [review]

**Review by awinogradov — APPROVED** [review]

File: src/components/RotatableTip/RotatableTip.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        };
+    }, [generateTip]);
+
+    const allTips = useMemo(() => {
+        return [...tips, ...(context ? contextTips[context] : [])];
+    }, [context, contextTips, tips]);
+
+    return (
+        <Tip title={tr('Pro tip!')} icon={<IconBulbOnOutline size={size} color={color} />} size={size}>
+            {allTips[Math.floor(Math.random() * allTips.length)](context)}
```
Comment (9teen90nine) [inline src/components/RotatableTip/RotatableTip.tsx:46]: You probably should cache index into a useMemo

File: src/components/RotatableTip/RotatableTip.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        </Tip>
+    );
+};
+
+export default RotatableTip;
```
Comment (9teen90nine) [inline src/components/RotatableTip/RotatableTip.tsx:51]: default export?

Comment (DenisVorop) [reply src/components/RotatableTip/RotatableTip.tsx:51]: that needed to use dynamic import

**Review by 9teen90nine — APPROVED** [review]

**Pull request merged**